### PR TITLE
Close requestor session

### LINKED ISSUE #10: Get warnings when exiting program
**Describe the bug**

When I exit my Discord bot, I get a lot of ResourceWarnings from unclosed sessions.

**To Reproduce**
1. Make reddit instance
2. Await a sleep.
3. Handle KeyboardInterrupt and instantiate new instance
4. Hit Ctrl + C
5. Wait 5 seconds
6. Hit Ctrl + C again

**Expected behavior**

Program exits without any warnings.

**Code/Logs**

Code: 
```python
import asyncpraw
import asyncio

reddit = asyncpraw.Reddit(<login info>)

async def main():
    await reddit.submission("asdfgh")
    await asyncio.sleep(1000)

if __name__ == '__main__':
    loop = asyncio.get_event_loop()
    try:
        loop.run_until_complete(main())
    except:
        reddit = asyncpraw.Reddit(<login info>)
        try:
            loop.run_until_complete(main())
        except:
            pass
```

Output: 
```
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x10c4f0c10>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x10c5346a0>, 0.68580921)]']
connector: <aiohttp.connector.TCPConnector object at 0x10c4f0c40>
```

**System Info**
 - OS: Mac OS 10.15.6
 - Python: 3.8.5
 - Async PRAW Version: 7.1.0

### PR DESCRIPTION
**Pull request by GerardRodes**
Fixes #10

## Feature Summary and Justification

A method to close the requestor session

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,5 @@
 _build/
 build/
 dist/
+.venv
+.vscode
```
Comment (PythonCoderAS) [inline .gitignore:13]:
```suggestion
```

Please do not add a .vscode folder. The .gitignore should be for universal things, such as build artifacts, but not preferences of IDEs. Use the `<base folder for asyncpraw>/.git/info/exclude` to exclude stuff locally without being pushed to remote..

**Review by PythonCoderAS — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (vikramaditya91) [reply .gitignore]: @PythonCoderAS  is it a convention followed in prawdev to avoid such things in .gitignore?

Comment (LilSpazJoekp) [reply .gitignore]: Things like `.vscode` and `.idea` should be in the user's global `.gitignore`. The project level `.gitignore` is for artifacts made by the project itself. See https://github.com/praw-dev/praw/pull/1160 for more info.

Comment (LilSpazJoekp) [reply .gitignore]: I am in favor for adding a `.venv` ignore. However, unless it is added to the main PRAW `.gitignore`, we'll leave it out of our `.gitignore` as well. I'd like to keep things standardized between the projects as much as possible.

File: asyncpraw/reddit.py [hunk 2]
```diff
@@ -119,6 +119,10 @@ def __enter__(self):
     def __exit__(self, *_args):
         """Handle the context manager close."""
 
+    async def close(self):
+        """Close open contexts."""
```
Comment (PythonCoderAS) [inline asyncpraw/reddit.py:123]:
```suggestion
        """Close the requestor."""
```

**Review by PythonCoderAS — CHANGES_REQUESTED** [review]

**Comment by LilSpazJoekp** [comment]
@GerardRodes If you could grant maintainer edit access, I can get this fixed up and merged in.

**Comment by GerardRodes** [comment]
> @GerardRodes If you could re-push your changes to your branch, it should allow you to reopen this PR. After that I'll get it fixed up and merged in. Thanks!

```dif
diff --git a/asyncpraw/reddit.py b/asyncpraw/reddit.py
index ce98a3bd..dc2f9b7e 100644
--- a/asyncpraw/reddit.py
+++ b/asyncpraw/reddit.py
@@ -119,10 +119,6 @@ class Reddit:
     def __exit__(self, *_args):
         """Handle the context manager close."""
 
+    async def close(self):
+        """Close open contexts."""
+        await self.requestor.close()
+
     def __init__(
         self,
         site_name: str = None,
@@ -225,7 +221,7 @@ class Reddit:
             )
 
         self._prepare_objector()
+        self.requestor = self._prepare_asyncprawcore(requestor_class, requestor_kwargs)
-        self._prepare_asyncprawcore(requestor_class, requestor_kwargs)
 
         self.auth = models.Auth(self, None)
         """An instance of :class:`.Auth`.
@@ -422,8 +418,6 @@ class Reddit:
         else:
             self._prepare_untrusted_asyncprawcore(requestor)
 
+        return requestor
+
     def _prepare_trusted_asyncprawcore(self, requestor):
         authenticator = TrustedAuthenticator(
             requestor,
```

this is the `diff` for some reason it doesn't recognizes changes between this branch and master

**Comment by LilSpazJoekp** [comment]
Yeah, I pushed the wrong branch and it brought it inline with master.

**Comment by LilSpazJoekp** [comment]
@GerardRodes Could you re-push that? You'll have to rebase your local version first.

**Comment by GerardRodes** [comment]
> @GerardRodes Could you re-push that? You'll have to rebase your local version first.

here it is https://github.com/praw-dev/asyncpraw/pull/25/

**Comment by LilSpazJoekp** [comment]
Could you close the other PR and reopen this one please? I'd like to keep everything together in one PR.

**Review by PythonCoderAS — APPROVED** [review]

**Pull request merged**